packages/discovery-react-components/src/components/SearchFacets/components/ShowMore/ShowMoreButton.tsx

```diff
@@ -25,11 +29,30 @@ export const ShowMoreButton: FC<ShowMoreButtonProps> = ({
   onClick,
   idSuffix,
   isCollapsed,
+  isShowAllMessage,
   messages
 }) => {
   return (
-    <Button kind="ghost" size="small" onClick={onClick} data-testid={`show-more-less-${idSuffix}`}>
-      {isCollapsed ? messages.collapsedFacetShowMoreText : messages.collapsedFacetShowLessText}
-    </Button>
+    <>
+      {isShowAllMessage ? (
+        <Button
+          kind="ghost"
+          size="small"
+          onClick={onClick}
+          data-testid={`show-more-less-${idSuffix}`}
+        >
+          {messages.collapsedFacetShowAllText}
+        </Button>
+      ) : (
+        <Button
+          kind="ghost"
+          size="small"
+          onClick={onClick}
+          data-testid={`show-more-less-${idSuffix}`}
+        >
+          {isCollapsed ? messages.collapsedFacetShowMoreText : messages.collapsedFacetShowLessText}
+        </Button>
+      )}
```

Comment:
You can define a function before return which would give you a message based on the conditions, and then you can simply render a button without duplicating it. 
Something like -
```

   const showMessage = () => {
    if (isShowAllMessage) {
      return messages.collapsedFacetShowAllText;
    } else if (isCollapsed) {
      return messages.collapsedFacetShowMoreText;
    }
    return messages.collapsedFacetShowLessText;
  };

  return (
    <Button kind="ghost" size="small" onClick={onClick} data-testid={`show-more-less-${idSuffix}`}>
      {showMessage}
    </Button>
  );
```